rdrf/rdrf/views/patient_view.py

```diff
@@ -539,22 +547,44 @@ def get_forms(self, request, registry_model, user, instance=None):
 
         return forms
 
-    def _section_hidden(self, user, registry, fieldlist):
+    def _exclude_hidden_fields(self, user, registry, fieldlist):
+        from rdrf.models.definition.models import DemographicFields
+        user_groups = [g.name for g in user.groups.all()]
+        hidden_fields = DemographicFields.objects.filter(
+            registry=registry, groups__name__in=user_groups, hidden=True, is_section=False
+        ).values_list('field', flat=True)
+        return list(set(fieldlist) - set(hidden_fields))
+
+    def _section_fields_hidden(self, user, registry, fieldlist):
```

Comment:
Shouldn't this be called `_all_section_fields_hidden`?
And why isn't the implementation just `len(self._exclude_hidden_fields(...)) == 0` ?

Reply:
Updated.